### ISSUE TITLE
invalid syntax in endheaders method definition for Python2.7

### ISSUE THREAD

**Issue by coreycb** [eventlet/eventlet#535]
```
Setting up python-eventlet (0.24.1-0ubuntu1) ...
  File "/usr/lib/python2.7/dist-packages/eventlet/green/http/client.py", line 1287
    def endheaders(self, message_body=None, *, encode_chunked=False):
                                             ^
SyntaxError: invalid syntax

dpkg: error processing package python-eventlet (--configure):
```

---------------------------------------

```
$ python3.7
Python 3.7.1 (default, Oct 21 2018, 08:03:53) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def endheaders(self, message_body=None, *, encode_chunked=False):
...   print("HELLO")
... 
>>> quit()
```

```
$ python3.6
Python 3.6.7 (default, Oct 21 2018, 08:08:16) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def endheaders(self, message_body=None, *, encode_chunked=False):
...   print("HELLO")
... 
>>> quit()
```

```
$ python2.7
Python 2.7.15+ (default, Nov 28 2018, 16:27:22) 
[GCC 8.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> def endheaders(self, message_body=None, *, encode_chunked=False):
  File "<stdin>", line 1
    def endheaders(self, message_body=None, *, encode_chunked=False):
                                             ^
SyntaxError: invalid syntax
>>> quit()
```

**Comment by coreycb** [comment]
https://stackoverflow.com/questions/2965271/forced-naming-of-parameters-in-python/14298976#14298976

**Comment by coreycb** [comment]
Here's another instance of this error:

```

```
Setting up python-eventlet (0.24.1-0ubuntu2~ubuntu19.04.1~ppa201812181504) ...
  File "/usr/lib/python2.7/dist-packages/eventlet/green/http/client.py", line 1300
    def request(self, method, url, body=None, headers={}, *,
                                                           ^
SyntaxError: invalid syntax
```

```

**Comment by nat-goodspeed** [comment]
Um, wow, you're right. It sure looks as though this problem dates back almost two years:

https://github.com/eventlet/eventlet/commit/7595a5a4#diff-34ee0f1767e598ba206f0c8c65e1c019R1284
https://github.com/eventlet/eventlet/commit/7595a5a4#diff-34ee0f1767e598ba206f0c8c65e1c019R1297

**Comment by nat-goodspeed** [comment]
Stylistically, the rest of eventlet source favors the Python 2 compatible approach of declaring `**kwds` and then using `kwds.pop('keyword_only_arg', default_value)`.

**Comment by nat-goodspeed** [comment]
@coreycb, I propose to create a pull request with relevant changes. Are you able to install from a github repository branch to test?

**Comment by jstasiak** [comment]
> Stylistically, the rest of eventlet source favors the Python 2 compatible approach of declaring `**kwds` and then using `kwds.pop('keyword_only_arg', default_value)`.

It'd be nice to defensively `assert not kwds, 'Unexpected keyword arguments: %s' % (kwds,)` or equivalent, so that typos in the client code are not silent if they occur in the future.

**Comment by nat-goodspeed** [comment]
Okay, that's what I was asking.

Both Python 2 and Python 3 raise `TypeError`, e.g.:

    TypeError: func() got an unexpected keyword argument 'jkl'

**Comment by nat-goodspeed** [comment]
Resolved by #536
